src/front/wgsl/lexer.rs

```diff
@@ -215,6 +228,32 @@ impl<'a> Lexer<'a> {
         }
     }
 
+    pub(super) fn expect_generic(&mut self, expected: Token<'a>) -> Result<(), Error<'a>> {
```

Comment:
can we refactor this into `fn expect_generic(&mut self, paren: char) -> ...`? Since it's only called on `Token::Paren`, specifying it seems redundant

Reply:
In this form it's consistent with `fn expect(&mut self, expected: Token<'a>)`, analogous to `next` and `next_generic`, and the implementation details are internal to the lexer with the only difference being if `true` or `false` are passed to `consume_token`; don't you think it's cleaner this way?

Reply:
It's a lot more code. We don't want to be generic where we don't have to. This could be a very specific thing called `expect_generic_paren` for clarity.

Reply:
It was only a lot more code because the match was duplicated from `expect` :) It should look much better now I think.

Reply:
Are you anticipating `expect_generic` to be used for anything *but* parens?
If not, then `expect_generic` doesn't need to work with any `Token`, and consequently the `describe_expect` isn't going to be needed (although I have nothing against it)

Reply:
I don't think it would be necessary for anything else. I just prefer to keep a consistent API 😄 I'll change it to `expect_generic_paren` for clarity then, it can always be changed in the future if needed.

Reply:
thank you! I don't want us to over-engineer this.
General rule is start to abstract once you have 2-3 repeating cases, not 1.